lib/plugins/mongoose-query/plugin/permissionsQuery.js

```
@@ -0,0 +1,119 @@
+import errors from 'errors'
+import _ from 'lodash'
+
+export default function attachStaticMethod(schema) {
+  schema.statics.permissionsQuery = permissionsQuery
+}
+
+function permissionsQuery(defaultQuery, permissionOptions, callback) {
+  const thisModule = this.modelName
+  if (!permissionOptions.fields || !permissionOptions.filters || validFilters(permissionOptions.filters)) {
+    const error = new errors.Http400Error({
+      message: 'Invalid permission options',
+      explanation: 'Input of permission query is not valid.'
+    })
```

We need to remove all instances of http errors from the module. It has no concept of a http request and should return a normal error or custom errors if we have those